### ISSUE TITLE
the building offered in the event has already been built

### ISSUE THREAD

**Issue by MrZorG33** [We-the-People-civ4col-mod/Mod#980]
in my test game the EVENTTRIGGER_BURNING_CORN_CHAMBER trigger worked. however, despite the presence of gold in the treasury, I cannot make the choice of the third option (subtract 1500 gold from the treasury and get Granary).
![Civ4ScreenShot0005](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/b31f3917-547a-4bbc-a95f-333b235d4f45)

and this happened due to the fact that the building proposed for construction (Granary) had already been built.
![Civ4ScreenShot0006](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/c3444bdb-8612-46f0-bc91-75d1a4214fea)

Is it possible to add to the trigger a check for the presence of the building proposed in the third option?

**Comment by Schmiddie1979** [comment]
I think in this case it is ok not to have the option - you already have the building. Ok, would be nice in this case the third option would not be shown...but I assume that would require a further Python function/code. Hence, I think we can live with that?

**Comment by MrZorG33** [comment]
yes we can live with it, it's not something that needs an instant fix. but it's better to fix it.

**Comment by Razonatair** [comment]
What's wrong with allowing players to view other potential options?

**Comment by MrZorG33** [comment]
@Razonatair There is nothing wrong. but this confuses the player. as an option, show a hint why this choice cannot be made.

**Comment by Schmiddie1979** [comment]
I have solved this issue. I have added a small Python function that checks if the city already has a granary. It the city has a granary the event will not trigger - i.e. it can trigger at a later time if a proper city is available.
